From b36e546ce18d00edc0480099a6b46b514c80e7d7 Mon Sep 17 00:00:00 2001
From: Dong Aisheng <b29396@freescale.com>
Date: Thu, 22 Jan 2015 21:48:07 +0800
Subject: [PATCH 0598/5242] MLK-10131 flexcan: fix wakeup unwork issue

commit  b0e0006f514d451d6cdb03af43d67e51f066e6e3 from
https://source.codeaurora.org/external/imx/linux-imx.git

Original code will always disable flexcan during suspend no matter whether
wakeup feature is enabled.
It's caused by merge issue when doing kernel upgrade.

Fix it by removing that disable code.

Signed-off-by: Dong Aisheng <b29396@freescale.com>
(cherry picked from commit 55c8670332973d3184816d3d16c2c3e07681c781)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/net/can/flexcan.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/can/flexcan.c b/drivers/net/can/flexcan.c
index 9e53082..6cff862 100644
--- a/drivers/net/can/flexcan.c
+++ b/drivers/net/can/flexcan.c
@@ -1576,7 +1576,7 @@ static int __maybe_unused flexcan_suspend(struct device *device)
 			err = flexcan_chip_disable(priv);
 		}
 	} else {
-		flexcan_chip_disable(priv);
+		err = flexcan_chip_disable(priv);
 	}
 	priv->can.state = CAN_STATE_SLEEPING;
 
-- 
1.7.9.5

